cmd/siklu/bank_management.c

```diff
@@ -96,8 +96,12 @@ struct software_bank_t* bank_management_handle_auto_switch(struct software_bank_
 		return bank;
 	} else if (boot_tries_left == 0) {
 		// out of tries, switch bank
-		bank = (bank == &first_bank) ? &second_bank : &first_bank;
-		siklu_fdt_setprop_string(fdt, "/", PROP_CURRENT_BANK, bank->bank_label);
+		if (bank != &first_bank && bank != &second_bank) {
+			printf("SIKLU_BOOT: Could not switch bank, wrong bank specified\n");
```

Comment:
maybe that should be the first thing on the function?

Reply:
with this idea we came back to the idea of extracting this block into function

Reply:
I don't exactly understand you response.

Reply:
I mean there was already discussion to extract this block into separate function that will make this check in the beggining

Reply:
Ok, then maybe we need to extract it?

Reply:
Do we want to split fdt write into 2 steps (bank and counter)? If we extract those 6 lines of code we will have to either do 2 re-writes or think on how to optimize that

Reply:
I agree with first Shmuel's comment.
bank is function param.  And it should be validated before all other things.

Reply:
done: split actions into 2 functions